fe/fe-core/src/main/java/org/apache/doris/clone/Rebalancer.java

```diff
@@ -0,0 +1,80 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.clone;
+
+import org.apache.doris.catalog.TabletInvertedIndex;
+import org.apache.doris.clone.TabletScheduler.PathSlot;
+import org.apache.doris.system.SystemInfoService;
+import org.apache.doris.task.AgentBatchTask;
+import org.apache.doris.thrift.TStorageMedium;
+
+import com.google.common.collect.Lists;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/*
+ * Rebalancer is responsible for
+ * 1. selectAlternativeTablets: selecting alternative tablets by one rebalance strategy,
+ * and return them to tablet scheduler(maybe contains the concrete moves, or maybe not).
+ * 2. createBalanceTask: given a tablet, try to create a clone task for this tablet.
+ * 3. getCachedSrcBackendId: if the rebalance strategy wants to delete the replica of a specified be,
+ * override this func.
+ * NOTICE:
+ * It may have a long interval between selectAlternativeTablets() & createBalanceTask(). So the concrete moves may be
+ * invalid when we createBalanceTask(), you should check the moves' validation.
+ */
+public abstract class Rebalancer {
```

Comment:
For a balance, the core strategy is how to choose which tablet, choose src backend, choose dest backend. But I don't see the interface about how to choose dest backend.

Reply:
Choosing dest be can be done before generating a clone task in `createBalanceTask`. The base class doesn't limit it.
For example:
The origin rebalance algo: `selectAlternativeTablets` just select a tablet, choose dest be in `createBalanceTask`
The algo which has a concrete move: `selectAlternativeTablets` will select a tablet and src be, dest be. In `createBalanceTask`, we should check the validation.

Reply:
> Choosing dest be can be done before generating a clone task in createBalanceTask. The base class doesn't limit it.

how to choose the dest be is core for balance strategy, if the base class doesn't limit it, what's the meaning of balance base class.

Reply:
In origin LoadBalancer class, both "Choose Source" and "Choose Destination" are done in `createBalanceTask`. 

@kangkaisen  may think that as a base class, these two interfaces need to be explicitly provided. But I think this is just a design pattern consideration, for the actual code implementation, there is little difference between the two. One is to implement the interfaces separately, and the other is to implement them in one interface.
The former has a clearer interface meaning, while the latter currently has relatively fewer code changes.